### PR TITLE
Facebook integration

### PR DESCRIPTION
**Pull request by pandilladivya**
**Integrated Facebook Sign In**

facebook_integration

The following is the screenshot of the log which shows the fetched data from Facebook API

screen shot 2017-06-05 at 12 21 35 am

### CODE REVIEW

**Comment by pandilladivya** [comment]
Kindly Review
@sandarumk @HimanshiJain 

🙂

File: app/src/main/java/vola/systers/com/volunteers_android/SignInActivity.java [hunk 1]
```diff
@@ -30,19 +38,22 @@
         GoogleApiClient.OnConnectionFailedListener {
 
     private static final String TAG = SignInActivity.class.getSimpleName();
-    private static final int RC_SIGN_IN = 007;
-
+    private static final int RC_GOOGLE_SIGN_IN = 007;
+    private static final int RC_FACEBOOK_SIGN_IN=64206;
```
Comment (vaibhavsingh97) [inline app/src/main/java/vola/systers/com/volunteers_android/SignInActivity.java]: This field can be move to integers.xml

**Comment by MeepyMay** [comment]
@pandilladivya Switch to 1.0 branch please. :)

File: app/src/main/java/vola/systers/com/volunteers_android/SignInActivity.java [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+            }
+
+            @Override
+            public void onCancel() {
+
+            }
+
+            @Override
+            public void onError(FacebookException error) {
+
```
Comment (pandilladivya) [inline app/src/main/java/vola/systers/com/volunteers_android/SignInActivity.java]: Yeah. I'll put a toast :)

File: app/src/main/res/layout/activity_signin.xml [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        android:text="@string/btn_signin_with_google"
+        android:textAppearance="@style/TextAppearance.AppCompat.Caption"
+        android:textColor="@color/text_white"
+        android:textStyle="bold"
+        android:drawableLeft="@drawable/google_plus"/>
+
+
+    <com.facebook.login.widget.LoginButton
+        xmlns:fb="http://schemas.android.com/apk/res-auto"
+        android:id="@+id/usersettings_fragment_login_button"
```
Comment (pandilladivya) [inline app/src/main/res/layout/activity_signin.xml]: okay.

File: app/src/main/res/layout/activity_signin.xml [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        android:layout_width="fill_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginLeft="30dp"
```
Comment (pandilladivya) [inline app/src/main/res/layout/activity_signin.xml]: Okay.

**Comment by pandilladivya** [comment]
@HimanshiJain Thanks for the review.
I've made changes.
Kindly review.
Thanks

File: app/src/main/res/values/styles.xml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        <item name="android:background">@color/facebook_signin</item>
+        <item name="android:layout_marginTop">10dp</item>
+        <item name="android:layout_marginBottom">10dp</item>
+        <item name="android:layout_gravity">center_horizontal</item>
+        <item name="android:textAppearance">@style/TextAppearance.AppCompat.Caption</item>
+        <item name="android:textColor">@color/text_white</item>
+        <item name="android:textStyle">bold</item>
+        <item name="android:drawableLeft">@drawable/facebook</item>
+        <item name="android:layout_height">200dp</item>
+
```
Comment (pandilladivya) [inline app/src/main/res/values/styles.xml]: Removed it.

**Comment by sandarumk** [comment]
As a general good practice please use meaningful commit messages. Commits like "Implemented Suggested changes" makes no sense. Please keep that in mind for future PRs

**Comment by HimanshiJain** [comment]
@pandilladivya Facebook integration is working. Good work. But as I said, the Google and Facebook Sign In button should be dimensionally same.

**Comment by HimanshiJain** [comment]
"Click here to skip this step" not working.

**Comment by pandilladivya** [comment]
Yes. This PR has facebook integration functionality and that is working in next PRs.
🙂

**Comment by HimanshiJain** [comment]
It is actually not working on any of those. And each time I want to test the working of a feature, I have to update the manifest to launch from a particular activity in subsequent PRs  because I can not login through Google or Facebook through those and even on the branch in which I can sign in manually it stays on the same activity. 

So it's actually fine if it isn't working here because probably you don't have any other activity. Make it working for the other branches.

**Comment by pandilladivya** [comment]
After this PR is merged. I'll have to merge these features on other branches too to resolve conflicts. And it will be easy for you to review too. 🙂

**Comment by HimanshiJain** [comment]
I agree. But this option needs to work right? So let us shift this discussion to the PR: https://github.com/systers/volunteers-android/pull/15

**Comment by pandilladivya** [comment]
@HimanshiJain @sandarumk  Is there anything more I can update in this PR?

File: app/src/main/java/vola/systers/com/volunteers_android/SignInActivity.java [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+    private ProgressDialog mProgressDialog;
+    CallbackManager callbackManager;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_signin);
+
+        LoginButton btnLogin=(LoginButton) findViewById(R.id.btn_fb_sign_in) ;
```
Comment (HimanshiJain) [inline app/src/main/java/vola/systers/com/volunteers_android/SignInActivity.java]: Use meaningful names for variables. We have three signin options. Something like googleLoginButton and so on?

File: app/src/main/java/vola/systers/com/volunteers_android/SignInActivity.java [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+
+    private void googleSignIn() {
+
+        Intent signInIntent = Auth.GoogleSignInApi.getSignInIntent(mGoogleApiClient);
+        startActivityForResult(signInIntent, RC_GOOGLE_SIGN_IN);
+
+
+    }
+
+    private void handlegoogleSignInResult(GoogleSignInResult result) {
```
Comment (HimanshiJain) [inline app/src/main/java/vola/systers/com/volunteers_android/SignInActivity.java]: shouldn't it be handleGoogleSignInResult?

Comment (pandilladivya) [inline app/src/main/java/vola/systers/com/volunteers_android/SignInActivity.java]: Yeah!
Will change :)
Anything else @HimanshiJain ?

Comment (HimanshiJain) [inline app/src/main/java/vola/systers/com/volunteers_android/SignInActivity.java]: I don't think so. Waiting for @sandarumk to review.

Comment (pandilladivya) [inline app/src/main/java/vola/systers/com/volunteers_android/SignInActivity.java]: @HimanshiJain Amended Naming changes to the previous commit. 😊

**Pull request merged**